python/lsst/ap/verify/workspace.py

```diff
@@ -29,13 +29,50 @@
 import lsst.daf.butler as dafButler
 
 
+# Workspace generic spec
+#
+# All Workspace classes must guarantee the existence of any subdirectories
+# inside the workspace. Directories corresponding to repositories do not need
+# to be initialized, since creating a valid repository usually requires
+# external information.
+#
+# Workspace classes may have additional properties specific to Gen 2 or Gen 3.
+# Keeping them in separate classes makes it easier to provide guarantees
+# without forcing awkward directory structures on users.
+#
+# (Spec implemented using duck-typing because AFAIK one can't create
+# abstract properties.)
```

Comment:
Do you mean "abstract _class_ properties or attributes"?  Because

```
@property
@abstractmethod
def my_abstract_property(self):
    raise NotImplementedError()
```
does indeed work, though I'm not sure an interface defined that way can be implemented with a regular attribute rather than an actual property, and I'm pretty sure it can't be implemented via a regular class attribute.

Anyhow, if that still doesn't quite work for you, I'd recommend just changing the language to "attribute" instead of "property".